python/adelphi/tests/unit/test_anonymize.py

```diff
@@ -0,0 +1,133 @@
+		self.assertTrue(index.name.startswith("idx"))
+		# check the index'es target column has been anonymized
+		self.assertTrue(index.index_options["target"].startswith("col"))
+
+	def test_anonymize_column(self):
+		schema = get_schema()
+		keyspace = schema.keyspaces[0]
+		table = keyspace.tables["my_table_0"]
+		column = table.columns["my_column_0"]
+		anonymize.anonymize_column(column)
+
+		self.assertTrue(column.name.startswith("col"))
+		self.assertEqual(column.cql_type, "ascii")
+
+	def test_anonymize_table(self):
+		schema = get_schema()
+		keyspace = schema.keyspaces[0]
+		table = keyspace.tables["my_table_0"]
+		anonymize.anonymize_table(table)
+		self.assertTrue(table.keyspace_name.startswith("ks"))
+		self.assertTrue(table.name.startswith("tbl"))
+
+		# check all columns at once
+		columns = [column.name for column in table.columns.values()]
+		self.assertPrefixes(columns, "col")
+
+		# check standard columns match their pk and ck definitions
+		self.assertEqual(columns[0], "col_0")
+		self.assertEqual(columns[1], "col_1")
+		self.assertEqual(columns[2], "col_2")
+		self.assertEqual(columns[3], "col_3")
+		self.assertEqual(table.partition_key[0].name, "col_0")
+		self.assertEqual(table.partition_key[1].name, "col_1")
+		self.assertEqual(table.clustering_key[0].name, "col_2")
+		self.assertEqual(table.clustering_key[1].name, "col_3")
+
+	def test_anonymize_keyspace(self):
+		schema = get_schema()
+		keyspace = schema.keyspaces[0]
+		anonymize.anonymize_keyspace(keyspace)
+		self.assertEqual(keyspace.name, "ks_0")
+		self.assertTrue(keyspace.durable_writes)
+		self.assertEqual(keyspace.replication_strategy.replication_factor_info.all_replicas, 1)
+
+		tables = [table.name for table in keyspace.tables.values()]
+		self.assertPrefixes(tables, "tbl")
+
+	def test_custom_indexes_are_removed(self):
+		schema = get_schema()
+		keyspace = schema.keyspaces[0]
+		tables = list(keyspace.tables.values())
+		table = tables[0]
+		
+		# ensure we have some custom indexes in the schema before anonymizing
+		custom_indexes = [index for index in table.indexes.values() if index.kind == "CUSTOM"]
+		self.assertTrue(len(custom_indexes) > 0)
+
+		anonymize.anonymize_keyspace(keyspace)
+		
+		# custom indexes should have been removed
+		custom_indexes = [index for index in table.indexes.values() if index.kind == "CUSTOM"]
+		self.assertTrue(len(custom_indexes) == 0)
+
+	def test_table_comment_is_removed(self):
+		schema = get_schema()
+		keyspace = schema.keyspaces[0]
+		tables = list(keyspace.tables.values())
+		table = tables[0]
+
+		self.assertTrue(table.options["comment"] == "Lorem ipsum")
+		anonymize.anonymize_table(table)
+		self.assertTrue(table.options["comment"] == "")
+
+	def test_anonymous_name_is_consistent(self):
+		name1 = get_name("test_column", COLUMN_PREFIX)
+		name2 = get_name("test_column", COLUMN_PREFIX)
+		name3 = get_name("another_column", COLUMN_PREFIX)
+		self.assertEqual(name1, name2)
+		self.assertNotEqual(name1, name3)
```

Comment:
Worth adding this at the end so that we can run this thing with a simple `python test_anonymize.py` (i.e. to avoid any explicit dependence on pytest):

```python
if __name__ == "__main__":
    unittest.main()
```

Reply:
done